src/client/components/three/stories/three.stories.tsx

```diff
@@ -0,0 +1,120 @@
+import { storiesOf } from '@storybook/react'
+import { IComputedValue } from 'mobx'
+import { reaction } from 'mobx'
+import { observable } from 'mobx'
+import { action } from 'mobx'
+import { computed } from 'mobx'
+import { disposeOnUnmount } from 'mobx-react'
+import { now } from 'mobx-utils'
+import { Component } from 'react'
+import * as React from 'react'
+import { Light } from 'three'
+import { Object3D } from 'three'
+import { Material } from 'three'
+import { Geometry } from 'three'
+import { Camera } from 'three'
+import { PointLight } from 'three'
+import { MeshPhongMaterial } from 'three'
+import { Mesh } from 'three'
+import { BoxGeometry } from 'three'
+import { PerspectiveCamera } from 'three'
+import { Scene } from 'three'
+
+import { Vector2 } from '../../../math/vector2'
+import { Vector3 } from '../../../math/vector3'
+import { Stage } from '../three'
+import { Canvas } from '../three'
+import { Three } from '../three'
+
+type BoxModel = { color: string, size: number, position: Vector3, rotation: Vector3 }
+
+class BoxVisualiser extends Component<{ animate?: boolean }> {
+  @observable
+  private model = {
+    boxes: [
+      { color: 'red', size: 1, position: Vector3.of(), rotation: Vector3.of() },
+      { color: 'green', size: 1, position: Vector3.of(), rotation: Vector3.of() },
+      { color: 'blue', size: 1, position: Vector3.of(), rotation: Vector3.of() },
+    ],
+  }
+
+  componentDidMount() {
+    this.update(0)
+    this.props.animate && disposeOnUnmount(this, reaction(() => now('frame'), this.update))
+  }
+
+  @action.bound
+  update(now: number) {
+    const t = 2 * Math.PI * now / (20 * 1000)
+    const n = this.model.boxes.length
+    this.model.boxes.forEach((box, i) => {
+      const position = Vector2.fromPolar(1, i * 2 * Math.PI / n + t)
+      box.position.set(position.x, position.y, 0)
+      box.rotation.set(Math.cos(3 * t + i), Math.cos(5 * t + i), Math.cos(7 * t + i))
+    })
+  }
+
+  render() {
+    return <Three createStage={this.createStage}/>
+  }
+
+  private createStage = (canvas: Canvas): IComputedValue<Stage> => {
+    const geometry = computed(() => this.boxGeometry())
+    const box = (box: BoxModel) => {
+      const material = computed(() => this.boxMaterial(box))
+      return computed(() => this.box(box, geometry.get(), material.get()))
+    }
+    const camera = computed(() => this.camera(canvas))
+    const light = computed(() => this.light(camera.get()))
+    const boxes = computed(() => this.model.boxes.map(b => box(b).get()))
+    const scene = computed(() => this.scene([...boxes.get(), light.get()]))
+    return computed(() => ({ camera: camera.get(), scene: scene.get() }))
```

Comment:
It is important to note that `createStage` returns a `IComputedValue<Stage>`, not just a `Stage`.

Specifically, the stage is not actually created until the computed is used. e.g.

```ts
const stage = createStage(canvas); // No three.js objects have been created yet. All we have done is create the computation tree.
stage.get(); // This is the call which recursively creates the three.js Scene and Camera objects and their children.
```

Having the entire scene being computed like this allows the render to:

1. react to any observable changes, and;
2. only require the computed functions whose dependencies have changes to re-compute.

This gives us the declarative efficiency we need.

Perhaps `createStage` is a misleading name then, but `createComputedStage` seems quite long...